pkg/skaffold/render/generate/generate.go

```diff
@@ -105,7 +101,49 @@ func (g *Generator) Generate(ctx context.Context) (manifest.ManifestList, error)
 		}
 		manifests.Append(out)
 	}
-	for _, nkPath := range nonKustomizePaths {
+
+	// Generate in-place hydrated kpt Manifests
+	kptPaths := excludeRemote(g.config.Kpt)
+	_, endTrace = instrumentation.StartTrace(ctx, "Render_expandGlobKptManifests")
+	kptPaths, err = util.ExpandPathsGlob(g.workingDir, kptPaths)
+	if err != nil {
+		event.DeployInfoEvent(fmt.Errorf("could not expand the glob kpt manifests: %w", err))
+		return nil, err
+	}
+	endTrace()
+	kptPathMap := make(map[string]bool)
+	for _, path := range kptPaths {
+		if dir, ok := isKptDir(path); ok {
+			kptPathMap[dir] = true
+		}
+	}
+	var rawManifestsFromKpt []string
+	for kPath := range kptPathMap {
+		// kpt manifests will be hydrated and stored in the subdir of the hydrated dir, where the subdir name
+		// matches the kPath dir name.
+		outputDir := filepath.Join(g.hydrationDir, filepath.Base(kPath))
+		_, endTrace := instrumentation.StartTrace(ctx, "Render_generateKptManifests")
+		cmd := exec.CommandContext(ctx, "kpt", "fn", "render", kPath,
+			fmt.Sprintf("--output=%v", outputDir))
+		cmd.Stderr = out
+		if err = util.RunCmd(cmd); err != nil {
+			endTrace(instrumentation.TraceEndError(err))
+			return nil, err
+		}
+		rawManifestsFromKpt = append(rawManifestsFromKpt, outputDir)
+	}
+
+	// Generate Raw Manifests
+	rawK8sPaths := excludeRemote(g.config.RawK8s)
```

Comment:
nit: `sourceManifests` instead of `rawK8sPaths`; `kptManifests` instead of `rawManifestsFromKpt`. The `raw` nature is implied and all the variables are local scope anyways.

Reply:
sg. Thanks for the suggestion.